C/crun/build_tarballs.jl

```diff
@@ -0,0 +1,54 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, BinaryBuilderBase, Pkg
+
+name = "crun"
+version = v"1.7.1"
+
+# Collection of sources required to complete build
+sources = [
+    ArchiveSource("https://github.com/containers/crun/releases/download/$(version)/crun-$(version).tar.xz",
+                  "bceade123d27ce31ab31bca14351e0cf4951e57b5b45de7a2dd3d512ef17912f")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+# crun requires a more recent libc than the ones provided by BB
+if [[ ${target} == *gnu* ]]; then
+    GLIBC_ARTIFACT_DIR=$(dirname $(dirname $(dirname $(realpath $prefix/usr/include/sched.h))))
+    rsync --archive ${GLIBC_ARTIFACT_DIR}/ /opt/${target}/${target}/sys-root/
+fi
+
+cd crun-*
+install_license COPYING
+
+./autogen.sh
+./configure --prefix=${prefix} --build=${MACHTYPE} --host=${target} \
+            --enable-shared --enable-dynamic \
+            --disable-seccomp --disable-systemd --disable-criu
+# disabled components are because of missing JLLs
+make -j${nproc}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter!(Sys.islinux, supported_platforms())
+filter!(p -> libc(p) != "musl", platforms)  # Musl_jll doesn't ship headers
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("crun", :crun),
+    LibraryProduct("libcrun", :libcrun)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("YAJL_jll"),
+    Dependency("libcap_jll"),
+    BuildDependency(PackageSpec(name = "Glibc_jll", version = v"2.17"))
+]
```

Comment:
Maybe do this only for `libc(p) == "gnu" && proc_family(p) == "Intel"`? No need to hack the build environment in the other cases

Reply:
It is done conditionally in the build script, right? By checking `if [[ ${target} == *gnu* ]]` (which I could make a little more strict). Doing this in Julia would mean splitting the `build_tarballs` invocations, which seems messy.

Reply:
I was suggesting filtering the dependency by platforms, with the `platforms` keyword argument to [`BuildDependency`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.BuildDependency)

Reply:
:TIL: again; let me try this out.